src/main/scala/QuartzSchedulerExtension.scala

```diff
@@ -54,11 +54,11 @@ class QuartzSchedulerExtension(system: ExtendedActorSystem) extends Extension {
    * Parses job and trigger configurations, preparing them for any code request of a matching job.
    * In our world, jobs and triggers are essentially 'merged'  - our scheduler is built around triggers
    * and jobs are basically 'idiot' programs who fire off messages.
-   *
-   * RECAST KEY AS UPPERCASE TO AVOID RUNTIME LOOKUP ISSUES
    */
-  var schedules: immutable.Map[String, QuartzSchedule] = QuartzSchedules(config, defaultTimezone)
-  val runningJobs: mutable.Map[String, JobKey] = mutable.Map.empty[String, JobKey]
+  val schedules: ConcurrentHashMap[String, QuartzSchedule] = new ConcurrentHashMap[String, QuartzSchedule](
```

Comment:
Instead of using `java.util.concurrent.ConcurrentHashMap`, could you use `scala.collection.concurrent.TrieMap`? This should give us the thread safety guarantees without all of the extra Java fluff.

Reply:
updated